crypto/crypto/src/hash.rs

```diff
@@ -62,16 +62,18 @@
 //! For any new structure `MyNewStruct` that needs to be hashed, it is recommended to simply
 //! use the derive macro [`CryptoHasher`](https://doc.rust-lang.org/reference/procedural-macros.html).
 //!
-//! ```ignore
-//! #[derive(CryptoHasher)]
-//! struct MyNewStruct {
-//!   ...
-//! }
+//! ```
+//! # use libra_crypto_derive::CryptoHasher;
+//! # use serde::Deserialize;
+//! #[derive(Deserialize, CryptoHasher)]
+//! #[serde(rename = "OptionalCustomSerdeName")]
+//! struct MyNewStruct { /*...*/ }
 //! ```
 //!
-//! The macro will define a hasher automatically called `MyNewStructHasher`, and pick a salt
-//! equal to the full module path + "::"  + structure name, i.e. if
-//! `MyNewStruct` is defined in `bar::baz::quux`, the salt will be `b"bar::baz::quux::MyNewStruct"`.
+//! The macro will define a hasher automatically called `MyNewStructHasher`, and derive a salt
+//! using the name of the type as seen by the Serde library. In the example above, this name
+//! was changed using the Serde parameter `rename`: the salt will be based on the value `OptionalCustomSerdeName`
+//! instead of the default name `MyNewStruct`.
```

Comment:
While we're speaking of "the macro", perhaps we want to desambiguate between `CryptoHasher` and `LCSCryptoHash` (and maybe even have a sentence about which does what)?